### PR TITLE
Run3-hcx243 Second step to make HCAL parameters to be read in DDD/DD4Hep

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Second step to read Hcal parameters using DDD as well as DD4Hep

#### PR validation:

Use the standard runTheMatrix.py

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 1]
```diff
@@ -285,58 +239,303 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
...
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          php.zxHE.emplace_back(std::abs(t.z()));
+          php.rhoxHE.emplace_back(t.Rho() * std::cos(t.phi()));
+          php.dyHE.emplace_back(dy * std::cos(t.phi()));
+          dx1 -= 0.5 * (t.Rho() - dy) * std::cos(t.phi()) * std::tan(10 * CLHEP::deg);
+          dx2 -= 0.5 * (t.Rho() + dy) * std::cos(t.phi()) * std::tan(10 * CLHEP::deg);
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]: tan(10 degrees) should be made into a constant:
```
static const double tan10deg = std::tan(10._deg);
```
We'd like to reduce usage of CLHEP.

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 2]
```diff
@@ -285,58 +239,303 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
...
+          php.dx1HE.emplace_back(-dx1);
+          php.dx2HE.emplace_back(-dx2);
+          php.layHE.emplace_back(lay);
+        }
+      }
+      if (copy[nsiz - 1] == 21 || copy[nsiz - 1] == 71) {
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]: 21 and 71 look like magic numbers that should be replaced with constants.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 3]
```diff
@@ -133,24 +91,24 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
                                    << t.Rho();
 #endif
       if (lay >= 0 && lay < 20) {
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]:
```suggestion
      if (lay >= 0 && lay < maxLayer_) {
```

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 4]
```diff
@@ -219,16 +177,14 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
                                    << t.z();
 #endif
       if (lay >= 0 && lay < 20) {
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]:
```suggestion
      if (lay >= 0 && lay < maxLayer_) {
```

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 5]
```diff
@@ -285,58 +241,303 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
...
+      dy = paras[1];
+      dz = paras[2];
+    }
+    if (idet == 3) {
+      // HB
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HCalGeom") << "HB " << fv.name() << " Shape "  //<< fv.shape()
+                                   << " Layer " << lay << " R " << t.Rho();
+#endif
+      if (lay >= 0 && lay < 20) {
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]:
```suggestion
      if (lay >= 0 && lay < maxLayer_) {
```

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 6]
```diff
@@ -285,58 +241,303 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
...
+#endif
+        }
+      }
+    } else if (idet == 4) {
+      // HE
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HCalGeom") << "HE " << fv.name() << " Shape "  //<< fv.shape()
+                                   << " Layer " << lay << " Z " << t.z();
+#endif
+      if (lay >= 0 && lay < 20) {
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]:
```suggestion
      if (lay >= 0 && lay < maxLayer_) {
```

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
@bsunanda thanks for addressing the issue of ```cout``` as suggested in a previous PR

**Comment by fabiocos** [comment]
+1

**Pull request merged**